src/MoBi.Core/Domain/Model/MoBiSimulation.cs

```diff
@@ -131,13 +131,22 @@ public override void UpdatePropertiesFrom(IUpdatable source, ICloneManager clone
 
       public void RemoveUsedObservedData(DataRepository dataRepository)
       {
-         
+         if (!UsesObservedData(dataRepository))
+            return;
 
+         var curveToRemove = Chart.Curves.FirstOrDefault(c => Equals(c.yData.Repository, dataRepository));
```

Comment:
Why First or default ?why are you assuming only one curve belongs to the repository? I think this is wrong

Reply:
true, was assuming one curve because of one dataRepository. Fixed to work for multiple.